<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="com.revature.controller.Controller_AdminTest" time="2.623" tests="18" errors="0" skipped="0" failures="18">
  <properties>
    <property name="jansi.mode" value="force"/>
    <property name="java.runtime.name" value="Java(TM) SE Runtime Environment"/>
    <property name="java.vm.version" value="21.0.2+13-LTS-58"/>
    <property name="sun.boot.library.path" value="C:\Program Files\Java\jdk-21\bin"/>
    <property name="maven.multiModuleProjectDirectory" value="D:\RevMart-1"/>
    <property name="java.vm.vendor" value="Oracle Corporation"/>
    <property name="java.vendor.url" value="https://java.oracle.com/"/>
    <property name="guice.disable.misplaced.annotation.check" value="true"/>
    <property name="path.separator" value=";"/>
    <property name="java.vm.name" value="Java HotSpot(TM) 64-Bit Server VM"/>
    <property name="sun.os.patch.level" value=""/>
    <property name="user.script" value=""/>
    <property name="user.country" value="IN"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="user.dir" value="D:\RevMart-1"/>
    <property name="java.vm.compressedOopsMode" value="32-bit"/>
    <property name="java.runtime.version" value="21.0.2+13-LTS-58"/>
    <property name="os.arch" value="amd64"/>
    <property name="java.io.tmpdir" value="C:\Users\shubh\AppData\Local\Temp\"/>
    <property name="line.separator" value="&#10;"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="user.variant" value=""/>
    <property name="stderr.encoding" value="UTF-8"/>
    <property name="os.name" value="Windows 11"/>
    <property name="classworlds.conf" value="D:\RevMart\.metadata\.plugins\org.eclipse.m2e.launching\launches\m2conf4706089207133663593.tmp"/>
    <property name="sun.jnu.encoding" value="Cp1252"/>
    <property name="stdout.encoding" value="UTF-8"/>
    <property name="java.library.path" value="C:\Program Files\Java\jdk-21\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\VMware\VMware Player\bin\;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\TDM-GCC-32\bin;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Common Files\Autodesk Shared\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Java\jdk1.8.0_311\bin;C:\xampp\php;C:\ProgramData\ComposerSetup\bin;C:\Windows\System32;C:\Program Files\nodejs\;C:\Program Files\Docker\Docker\resources\bin;C:\Program Files\MySQL\MySQL Server 8.0\bin;C:\Program Files\Git\cmd;D:\Downloads\apache-maven-3.9.8-bin\apache-maven-3.9.8\bin;C:\Users\shubh\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\shubh\AppData\Local\Programs\Python\Python39\;C:\MinGW\bin;C:\Users\shubh\AppData\Local\Microsoft\WindowsApps;C:\laragon\bin;C:\Users\shubh\AppData\Roaming\Composer\vendor\bin;C:\Users\shubh\AppData\Local\GitHubDesktop\bin;C:\Users\shubh\AppData\Roaming\npm;C:\Program Files\MongoDB\Server\6.0\bin;C:\Users\shubh\AppData\Local\Programs\mongosh\;C:\Users\shubh\AppData\Local\Programs\mongosh;C:\Users\shubh\AppData\Local\Programs\Microsoft VS Code\bin;;C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2023.3.4\bin;;."/>
    <property name="jdk.debug" value="release"/>
    <property name="java.class.version" value="65.0"/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers"/>
    <property name="os.version" value="10.0"/>
    <property name="user.home" value="C:\Users\shubh"/>
    <property name="user.timezone" value="Asia/Calcutta"/>
    <property name="file.encoding" value="UTF-8"/>
    <property name="java.specification.version" value="21"/>
    <property name="user.name" value="shubh"/>
    <property name="java.class.path" value="C:\Users\shubh\.p2\pool\plugins\org.eclipse.m2e.maven.runtime_3.9.700.20240602-2313\jars\plexus-classworlds-2.8.0.jar"/>
    <property name="java.vm.specification.version" value="21"/>
    <property name="sun.arch.data.model" value="64"/>
    <property name="sun.java.command" value="org.codehaus.plexus.classworlds.launcher.Launcher -Ppom.xml -B -Dstyle.color=always clean install"/>
    <property name="java.home" value="C:\Program Files\Java\jdk-21"/>
    <property name="user.language" value="en"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="java.vm.info" value="mixed mode, sharing"/>
    <property name="java.version" value="21.0.2"/>
    <property name="native.encoding" value="Cp1252"/>
    <property name="java.vendor" value="Oracle Corporation"/>
    <property name="maven.home" value="D:\RevMart-1\EMBEDDED"/>
    <property name="style.color" value="always"/>
    <property name="file.separator" value="\"/>
    <property name="java.version.date" value="2024-01-16"/>
    <property name="java.vendor.url.bug" value="https://bugreport.java.com/bugreport/"/>
    <property name="sun.io.unicode.encoding" value="UnicodeLittle"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="m2e.build.project.data.socket.port" value="56997"/>
    <property name="sun.cpu.isalist" value="amd64"/>
  </properties>
  <testcase name="com.revature.controller.Controller_AdminTest.testSaveCategory_InternalServerError" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.existCategory(String)&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.existCategory(String)" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testSaveCategory_InternalServerError(Controller_AdminTest.java:143)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testUpdateUserAccountStatus_Failure" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.UserService.updateAccountStatus(java.lang.Integer, java.lang.Boolean)&quot; because &quot;this.userService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.UserService.updateAccountStatus(java.lang.Integer, java.lang.Boolean)" because "this.userService" is null
	at com.revature.controller.Controller_AdminTest.testUpdateUserAccountStatus_Failure(Controller_AdminTest.java:395)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testUpdateUserAccountStatus_Success" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.UserService.updateAccountStatus(java.lang.Integer, java.lang.Boolean)&quot; because &quot;this.userService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.UserService.updateAccountStatus(java.lang.Integer, java.lang.Boolean)" because "this.userService" is null
	at com.revature.controller.Controller_AdminTest.testUpdateUserAccountStatus_Success(Controller_AdminTest.java:376)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testSaveCategory_CategoryAlreadyExists" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.existCategory(String)&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.existCategory(String)" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testSaveCategory_CategoryAlreadyExists(Controller_AdminTest.java:124)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testSaveProduct_Failure" classname="com.revature.controller.Controller_AdminTest" time="1.935">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.saveProduct(com.revature.model.Product)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.saveProduct(com.revature.model.Product)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testSaveProduct_Failure(Controller_AdminTest.java:226)
</failure>
    <system-err><![CDATA[WARNING: A Java agent has been loaded dynamically (C:\Users\shubh\.m2\repository\net\bytebuddy\byte-buddy-agent\1.14.19\byte-buddy-agent-1.14.19.jar)
WARNING: If a serviceability tool is in use, please run with -XX:+EnableDynamicAgentLoading to hide this warning
WARNING: If a serviceability tool is not in use, please run with -Djdk.instrument.traceUsage for more information
WARNING: Dynamic loading of agents will be disallowed by default in a future release
]]></system-err>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testDeleteCategory_Success" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.deleteCategory(int)&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.deleteCategory(int)" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testDeleteCategory_Success(Controller_AdminTest.java:158)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testUpdateProduct_Success" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.updateProduct(com.revature.model.Product, org.springframework.web.multipart.MultipartFile)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.updateProduct(com.revature.model.Product, org.springframework.web.multipart.MultipartFile)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testUpdateProduct_Success(Controller_AdminTest.java:300)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testDeleteCategory_Failure" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.deleteCategory(int)&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.deleteCategory(int)" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testDeleteCategory_Failure(Controller_AdminTest.java:172)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testDeleteProduct_Failure" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.deleteProduct(java.lang.Integer)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.deleteProduct(java.lang.Integer)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testDeleteProduct_Failure(Controller_AdminTest.java:258)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testLoadAddProduct_Success" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.getAllCategory()&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.getAllCategory()" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testLoadAddProduct_Success(Controller_AdminTest.java:89)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testGetAllUsers_RoleAdmin" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.UserService.getUsers(String)&quot; because &quot;this.userService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.UserService.getUsers(String)" because "this.userService" is null
	at com.revature.controller.Controller_AdminTest.testGetAllUsers_RoleAdmin(Controller_AdminTest.java:356)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testSaveProduct_Success" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.saveProduct(com.revature.model.Product)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.saveProduct(com.revature.model.Product)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testSaveProduct_Success(Controller_AdminTest.java:195)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testDeleteProduct_Success" classname="com.revature.controller.Controller_AdminTest" time="0">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.deleteProduct(java.lang.Integer)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.deleteProduct(java.lang.Integer)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testDeleteProduct_Success(Controller_AdminTest.java:242)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testGetAllUsers_RoleUser" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.UserService.getUsers(String)&quot; because &quot;this.userService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.UserService.getUsers(String)" because "this.userService" is null
	at com.revature.controller.Controller_AdminTest.testGetAllUsers_RoleUser(Controller_AdminTest.java:338)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testGetAllOrders" classname="com.revature.controller.Controller_AdminTest" time="0.562">
    <failure message="&#10;Misplaced or misused argument matcher detected here:&#10;&#10;-&gt; at com.revature.controller.Controller_AdminTest.testSaveProduct_Failure(Controller_AdminTest.java:226)&#10;-&gt; at com.revature.controller.Controller_AdminTest.testUpdateProduct_Success(Controller_AdminTest.java:300)&#10;-&gt; at com.revature.controller.Controller_AdminTest.testUpdateProduct_Success(Controller_AdminTest.java:300)&#10;-&gt; at com.revature.controller.Controller_AdminTest.testSaveProduct_Success(Controller_AdminTest.java:195)&#10;&#10;You cannot use argument matchers outside of verification or stubbing.&#10;Examples of correct usage of argument matchers:&#10;    when(mock.get(anyInt())).thenReturn(null);&#10;    doThrow(new RuntimeException()).when(mock).someVoidMethod(any());&#10;    verify(mock).someMethod(contains(&quot;foo&quot;))&#10;&#10;This message may appear after an NullPointerException if the last matcher is returning an object &#10;like any() but the stubbed method signature expect a primitive argument, in this case,&#10;use primitive alternatives.&#10;    when(mock.get(any())); // bad use, will raise NPE&#10;    when(mock.get(anyInt())); // correct usage use&#10;&#10;Also, this error might show up because you use argument matchers with methods that cannot be mocked.&#10;Following methods *cannot* be stubbed/verified: final/private/equals()/hashCode().&#10;Mocking methods declared on non-public parent classes is not supported.&#10;" type="org.mockito.exceptions.misusing.InvalidUseOfMatchersException">org.mockito.exceptions.misusing.InvalidUseOfMatchersException: 
Misplaced or misused argument matcher detected here:
-> at com.revature.controller.Controller_AdminTest.testSaveProduct_Failure(Controller_AdminTest.java:226)
-> at com.revature.controller.Controller_AdminTest.testUpdateProduct_Success(Controller_AdminTest.java:300)
-> at com.revature.controller.Controller_AdminTest.testUpdateProduct_Success(Controller_AdminTest.java:300)
-> at com.revature.controller.Controller_AdminTest.testSaveProduct_Success(Controller_AdminTest.java:195)
You cannot use argument matchers outside of verification or stubbing.
Examples of correct usage of argument matchers:
    when(mock.get(anyInt())).thenReturn(null);
    doThrow(new RuntimeException()).when(mock).someVoidMethod(any());
    verify(mock).someMethod(contains("foo"))
This message may appear after an NullPointerException if the last matcher is returning an object 
like any() but the stubbed method signature expect a primitive argument, in this case,
use primitive alternatives.
    when(mock.get(any())); // bad use, will raise NPE
    when(mock.get(anyInt())); // correct usage use
Also, this error might show up because you use argument matchers with methods that cannot be mocked.
Following methods *cannot* be stubbed/verified: final/private/equals()/hashCode().
Mocking methods declared on non-public parent classes is not supported.

	at com.revature.controller.Controller_AdminTest.testGetAllOrders(Controller_AdminTest.java:412)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testEditProduct" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.ProductService.getProductById(java.lang.Integer)&quot; because &quot;this.productService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.ProductService.getProductById(java.lang.Integer)" because "this.productService" is null
	at com.revature.controller.Controller_AdminTest.testEditProduct(Controller_AdminTest.java:275)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testLoadAddProduct_NoCategories" classname="com.revature.controller.Controller_AdminTest" time="0.001">
    <failure message="Cannot invoke &quot;com.revature.service.CategoryService.getAllCategory()&quot; because &quot;this.categoryService&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.service.CategoryService.getAllCategory()" because "this.categoryService" is null
	at com.revature.controller.Controller_AdminTest.testLoadAddProduct_NoCategories(Controller_AdminTest.java:103)
</failure>
  </testcase>
  <testcase name="com.revature.controller.Controller_AdminTest.testUpdateProduct_InvalidDiscount" classname="com.revature.controller.Controller_AdminTest" time="0.12">
    <failure message="Cannot invoke &quot;com.revature.controller.Controller_Admin.updateProduct(com.revature.model.Product, org.springframework.web.multipart.MultipartFile, jakarta.servlet.http.HttpSession, org.springframework.ui.Model)&quot; because &quot;this.adminController&quot; is null" type="java.lang.NullPointerException">java.lang.NullPointerException: Cannot invoke "com.revature.controller.Controller_Admin.updateProduct(com.revature.model.Product, org.springframework.web.multipart.MultipartFile, jakarta.servlet.http.HttpSession, org.springframework.ui.Model)" because "this.adminController" is null
	at com.revature.controller.Controller_AdminTest.testUpdateProduct_InvalidDiscount(Controller_AdminTest.java:326)
</failure>
  </testcase>
</testsuite>